src/core/opamSystem.ml

```diff
@@ -754,10 +795,10 @@ let install ?(warning=default_install_warning) ?exec src dst =
       in
       copy_file_aux ~src ~dst ();
       if cygcheck then
-        match OpamStd.Sys.is_cygwin_variant dst with
+        match OpamStd.Sys.get_windows_executable_variant dst with
           `Native ->
             ()
-        | `Cygwin ->
+        | `Cygwin | `Msys2 ->
             warning dst `Cygwin
         | `CygLinked ->
             warning dst `Cygwin_libraries
```

Comment:
```suggestion
        (`Cygwin | `Msys2 | `Tainted _) as code ->
            warning dst code
```
with corresponding updated constructors and messages in the `default_install_warning` function.